packages/utils/src/numbers/decimals.test.ts

```diff
@@ -87,6 +115,32 @@ describe('utils - decimals', () => {
       expect(toBigInt(0.78, 12)).toBe(780_000_000_000n);
       expect(toBigInt(0.372, 12)).toBe(372_000_000_000n);
       expect(toBigInt(0.372209875392, 12)).toBe(372_209_875_392n);
+      expect(toBigInt('4756.50334457721117752', 18)).toBe(
+        4_756_503_344_577_211_177_520n,
+      );
+      expect(toBigInt('66712.0367386073069948646', 18)).toBe(
+        66_712_036_738_607_306_994_864n,
+      );
+    });
+  });
+
+  describe('hasDecimalOverflow', () => {
+    it('should return true if the value has more decimals that allowed', () => {
+      expect(hasDecimalOverflow('66712.0367386073069948646', 18)).toBe(true);
+      expect(hasDecimalOverflow('66712.0367386073069948645', 18)).toBe(true);
+      expect(hasDecimalOverflow('12.036073069948646', 12)).toBe(true);
+      expect(hasDecimalOverflow('12.0360730692110', 12)).toBe(true);
+      expect(hasDecimalOverflow('12.036073069', 12)).toBe(false);
+      expect(hasDecimalOverflow('12.036073069210', 12)).toBe(false);
+      expect(hasDecimalOverflow('66712.036738607306994864', 18)).toBe(false);
```

Comment:
Should these 2 cases go to the second `it`?

Reply:
Yes actually :)
done